src/lib/sql/sql_translator.cpp

```diff
@@ -1834,15 +1835,32 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_import(const hsql::Im
   // so on. Anyway, we would need to decouple data loading and storing the table and have to load metadata or even the
   // whole table when translating the SQL statement.
   AssertInput(!import_statement.whereClause, "Predicates on imported files are not supported.");
+
+  if (!import_statement.encoding) {
+    return ImportNode::make(import_statement.tableName, import_statement.filePath,
+                            import_type_to_file_type(import_statement.type));
+  }
+
+  auto file_encoding =
+      magic_enum::enum_cast<EncodingType>(std::string{import_statement.encoding}, magic_enum::case_insensitive);
+  AssertInput(file_encoding.has_value(), "Unknown encoding type '" + std::string{import_statement.encoding} + "'.");
   return ImportNode::make(import_statement.tableName, import_statement.filePath,
-                          import_type_to_file_type(import_statement.type));
+                          import_type_to_file_type(import_statement.type), file_encoding.value());
 }
 
 // NOLINTNEXTLINE: while this particular method could be made static, others cannot.
 std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_export(const hsql::ExportStatement& export_statement) {
   auto sql_identifier_resolver = std::make_shared<SQLIdentifierResolver>();
   auto lqp = std::shared_ptr<AbstractLQPNode>{};
 
+  if (export_statement.encoding) {
+    AssertInput(false, "Encoding '" + std::string{export_statement.encoding} +
+                           "' was specified"
+                           " but a set output encoding is not yet supported."
+                           "To achieve the same result, export the table once, "
+                           "then load using the wished encoding and export again.");
+  }
+
```

Comment:
```c++
if (export_statement.encoding) {
  FailInput(...);
}
```
or just
```
AssertInput(!export_statement.encoding, ...);
```

I would also just be more brief here, sth like: "Encoding for table export is not supported. You can choose an encoding when importing tables."